### ISSUE TITLE
[Feature] Improve control plane performance

### ISSUE THREAD

**Issue by aolemila** [sgl-project/sglang-jax#64]
### Motivation

Current time cost of TPU computation is less than control plane, so improving control plane performance is important. Control plane consists of tokenizer, scheduler and other preparation work. Supporting overlap scheduler and regulating the numpy and jax.numpy usage is known to improve the performance. Here we focus on these two aspects to take the first step to improve control plane performance.

### Modification

- Support overlap in scheduler.
- Regulate the usage of numpy and jax.numpy in ScheduleBatch, ModelWorkerBatch, ForwardBatch, MemoryPool and PrefixCache
- Add jit for compute_logits and sample, both of them do not support jit with return_logprob
- ...

### Command

```bash
# launch server

JAX_COMPILATION_CACHE_DIR=/tmp/jit_cache  python3 -u -m sgl_jax.launch_server --model-path Qwen/Qwen-7B-Chat --trust-remote-code  --dist-init-addr=0.0.0.0:10011 --nnodes=1  --tp-size=4 --device=tpu --random-seed=3 --node-rank=0 --mem-fraction-static=0.4 --chunked-prefill-size=8192 --download-dir=/tmp --dtype=bfloat16 --jax-precompile-decode-bs-paddings 64 --max-running-requests 64 --skip-server-warmup --attention-backend=fa --jax-precompile-prefill-token-paddings 8192 --page-size=64

# do profile

curl -X POST 'http://127.0.0.1:30000/start_profile' -d '{"output_dir": {profile_dir}, "num_steps": 5}' -H 'Content-Type: application/json'

# benchmark

python3 -m sgl_jax.bench_serving --backend sgl-jax --dataset-name random --num-prompts 10 --random-input 512 --random-output 128 --random-range-ratio 1 --warmup-requests 0

```

### Perf Comparison

**main-662364bd3289de5c2a257df5b045c1a0ca17151e**

**current branch**

**Comment by JamesBrianD** [comment]
The control plane performance has been improved to a relatively fast speed, with a decoding interval overhead of less than 1.5ms. Close the issue first.
